lib/util/eyeglass_modules.js

```diff
@@ -0,0 +1,283 @@
+};
+
+EyeglassModules.prototype.find = function(name) {
+  return getFinalModule.call(this, name);
+};
+
+EyeglassModules.prototype.findByPath = function(dir) {
+  return this.list.find(function(mod) {
+    if (mod.path === dir) {
+      return getFinalModule.call(this, mod.name);
+    }
+  }.bind(this));
+};
+
+// this is mostly for debugging purposes
+EyeglassModules.prototype.getGraph = function() {
+  var hierarchy = getHierarchy(this.tree);
+  hierarchy.label = getDecoratedRootName.call(this);
+  return require("archy")(hierarchy);
+};
+
+function getDecoratedRootName() {
+  return ":root" + ((this.projectName) ? "(" + this.projectName + ")" : "");
+}
+
+function getHierarchyNodes(branch) {
+  if (branch) {
+    return Object.keys(branch).map(function(name) {
+      return getHierarchy(branch[name]);
+    });
+  }
+}
+
+function getHierarchy(branch) {
+  if (branch) {
+    return {
+      label: branch.name,
+      nodes: getHierarchyNodes(branch.dependencies)
+    };
+  }
+}
+
+function pruneModuleTree(moduleTree) {
+  var finalModule = moduleTree.name && getFinalModule.call(this, moduleTree.name);
+  var branch = {
+    name: finalModule && finalModule.name || moduleTree.name,
+    path: finalModule && finalModule.path || moduleTree.path
+  };
+  var dependencies = moduleTree.dependencies;
+  if (dependencies) {
+    branch.dependencies = Object.keys(dependencies).reduce(function(tree, name) {
+      tree[name] = pruneModuleTree.call(this, dependencies[name]);
+      return tree;
+    }.bind(this), {});
+  }
+  return branch;
+}
+
+function findBranchesByPath(modules, match) {
+  if (modules) {
+    return Object.keys(modules).reduce(function(branches, name) {
+      var mod = modules[name];
+      if (mod.path === match) {
+        branches.push(mod);
+      }
+      if (mod.dependencies) {
+        branches.push.apply(branches, findBranchesByPath(mod.dependencies, match));
+      }
+      return branches;
+    }, []);
+  }
+  return [];
+}
+
+function canAccessModule(name, origin) {
+  if (origin === true) {
+    return true;
+  }
+  origin = packageUtils.findNearestPackage(origin);
+  var branches = findBranchesByPath({
+    dependencies: this.tree
+  }, origin);
+  return branches.some(function(branch) {
+    if (branch.name === name || branch.dependencies[name]) {
+      return true;
+    }
+  });
+}
+
+function flattenModules(branch, registry) {
+  if (branch) {
+    // if the branch itself is a module, add it...
+    if (branch.isEyeglassModule) {
+      registry = registry || {};
+      registry[branch.name] = registry[branch.name] || [];
+      registry[branch.name].push(branch);
+    }
+
+    var dependencies = branch.dependencies;
+
+    if (dependencies) {
+      var result = Object.keys(dependencies).reduce(function(registry, name) {
+        return flattenModules(dependencies[name], registry);
+      }, registry || {});
+      return result;
+    }
+  }
+
+  return registry;
+}
+
+function checkVersions(versions, latest) {
```

Comment:
`checkVersions` seems to return incompatible versions. Either the fn name or the return object should probably change.